konnectivity-client/pkg/client/client.go

```diff
@@ -175,6 +190,8 @@ func (t *grpcTunnel) serve(c clientConn) {
 				case <-timer.C:
 					klog.ErrorS(fmt.Errorf("timeout"), "readTimeout has been reached, the grpc connection to the proxy server will be closed", "connectionID", conn.connID, "readTimeoutSeconds", t.readTimeoutSeconds)
 					return
+				case <-tunnelCtx.Done():
+					klog.V(1).InfoS("Tunnel has been closed, the grpc connection to the proxy server will be closed", "connectionID", conn.connID)
```

Comment:
(coming back from a long break and just going through PRs I missed)

Wondering why we don't return early in this specific case when the tunnel context gets cancelled.

Reply:
Was playing around with a unit test to see if a potential leak was reproducible: https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/399